src/configure/clients/devOps/azureDevOpsClient.ts

```diff
@@ -84,46 +160,42 @@ export class AzureDevOpsClient {
+
+        return this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: url,
+            method: "POST",
             headers: {
-                "Accept": "application/json;api-version=5.0-preview.1;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true",
-                "Content-Type": "application/json"
+                "Accept": "application/json;api-version=5.0-preview.7;"
             },
+            body: buildDefinition,
+            serializationMapper: null,
+            deserializationMapper: null
+        });
+    }
+
+    public async queueBuild(organizationName: string, build: Build): Promise<any> {
+        let url = await this.getBaseOrgUrl(organizationName, "tfs");
+        url = `${url}/${build.project.id}/_apis/build/builds`;
+
+        return this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: url,
             method: "POST",
-            body: {
-                "contributionIds": [
-                    "ms.vss-build-web.create-and-run-pipeline-data-provider"
-                ],
-                "dataProviderContext": {
-                    "properties": {
-                        "connectionId": inputs.sourceRepository.serviceConnectionId, //GitHub endpoint id
-                        "sourceProvider": inputs.sourceRepository.repositoryProvider,
-                        "repositoryId": inputs.sourceRepository.repositoryId,
-                        "repositoryName": inputs.sourceRepository.repositoryName,
-                        "branch": inputs.sourceRepository.branch,
-                        "sourceBranch": inputs.sourceRepository.branch,
-                        "path": inputs.pipelineParameters.pipelineFilePath,
-                        "queue": "Hosted Ubuntu 1604",
-                        "commitId": inputs.sourceRepository.commitId,
-                        "commitDescriptorName": "Set up CI/CD with Azure Pipelines",
-                        "sourcePage": {
-                            "routeValues": {
-                                "project": inputs.projectName
-                            }
-                        }
-                    }
-                }
+            headers: {
+                "Accept": "application/json;api-version=5.2-preview.5;"
             },
-            deserializationMapper: null,
-            serializationMapper: null
+            body: build,
+            serializationMapper: null,
+            deserializationMapper: null
         });
     }
 
-    public getBaseOrgUrl(organizationName: string, service: string): string {
+    public async getBaseOrgUrl(organizationName: string, service: string): Promise<string> {
         if (!this.lastAccessedOrganization || this.lastAccessedOrganization.accountName !== organizationName) {
-            this.lastAccessedOrganization = this.organizationMap.find((element) => {
+            let organizations = await this.listOrgPromise;
```

Comment:
Can we not hardcode the url based on org name, rather than making async call

Reply:
@hiyadav  Was there any specific reason to get the url this way rather than hardcoding to dev.azure.com